watch.ts

```diff
@@ -0,0 +1,48 @@
+import request = require('request');
+import { LineStream } from 'byline';
+import { KubeConfig } from '@kubernetes/typescript-node';
+
+export class Watch {
+    'config': KubeConfig;
+
+    public constructor(config: KubeConfig) {
+        this.config = config;
+    }
+
+    public watch(path: string, queryParams: any, callback: (phase: string, obj: any) => void, done: (err: any) => void): any {
```

Comment:
maybe we'd be better off returning a `Promise` from this func?

Reply:
I don't love returning a `Promise` here. The basic problem is this:

1) I want to return an object that enables someone to cancel the watch.  Promises aren't really cancelable.

2) I want to return a Promise that enables the user to determine when the watch is done.

Because there is no multi-value return in Typescript, it feels like passing a `done` callback and returning a cancelable object makes the most sense.